feature/platform/transceiver/tests/zr_supply_voltage_test/zr_supply_voltage_test.go

```diff
@@ -0,0 +1,125 @@
+	voltageSample := pStream.Next(t)
+	if voltageSample == nil {
+		t.Fatalf("Voltage telemetry %s was not streamed in the most recent subscription interval", path)
+	}
+	voltageVal, ok := voltageSample.Val()
+	if !ok {
+		t.Fatalf("Voltage %q telemetry is not present", voltageSample)
+	}
+	// Check voltage return value of correct type
+	if reflect.TypeOf(voltageVal).Kind() != reflect.Float64 {
+		t.Fatalf("Return value is not type float64")
+	}
+	t.Logf("Voltage sample value %s: %v", path, voltageVal)
+	return voltageVal
+}
+
+func TestZrSupplyVoltage(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	for _, port := range []string{"port1", "port2"} {
+		t.Run(fmt.Sprintf("Port:%s", port), func(t *testing.T) {
+			dp := dut.Port(t, port)
+
+			gnmi.Await(t, dut, gnmi.OC().Interface(dp.Name()).OperStatus().State(), intUpdateTime, oc.Interface_OperStatus_UP)
+
+			// Derive transceiver names from ports.
+			tr := gnmi.Get(t, dut, gnmi.OC().Interface(dp.Name()).Transceiver().State())
+			component := gnmi.OC().Component(tr)
+
+			outputPower := gnmi.Get(t, dut, component.OpticalChannel().TargetOutputPower().State())
+			if outputPower != targetOutputPower {
+				t.Fatalf("Output power does not match output power, got: %v want :%v", outputPower, targetOutputPower)
+			}
+
+			frequency := gnmi.Get(t, dut, component.OpticalChannel().Frequency().State())
+			if frequency != targetFrequency {
+				t.Fatalf("Frequency does not match frequency, got: %v want :%v", frequency, frequency)
+			}
+
+			streamInst := samplestream.New(t, dut, component.Transceiver().SupplyVoltage().Instant().State(), samplingInterval)
+			defer streamInst.Close()
+			streamAvg := samplestream.New(t, dut, component.Transceiver().SupplyVoltage().Avg().State(), samplingInterval)
+			defer streamAvg.Close()
+			streamMin := samplestream.New(t, dut, component.Transceiver().SupplyVoltage().Min().State(), samplingInterval)
+			defer streamMin.Close()
+			streamMax := samplestream.New(t, dut, component.Transceiver().SupplyVoltage().Max().State(), samplingInterval)
+			defer streamMax.Close()
+
+			volInst := verifyVoltageValue(t, dut, streamInst, "Instant")
+			t.Logf("Port %s instant voltage: %v", dp.Name(), volInst)
+			volAvg := verifyVoltageValue(t, dut, streamAvg, "Avg")
+			t.Logf("Port %s average voltage: %v", dp.Name(), volAvg)
+			volMin := verifyVoltageValue(t, dut, streamMin, "Min")
+			t.Logf("Port %s minimum voltage: %v", dp.Name(), volMin)
+			volMax := verifyVoltageValue(t, dut, streamMax, "Max")
+			t.Logf("Port %s maximum voltage: %v", dp.Name(), volMax)
+
+			if volAvg >= volMin && volAvg <= volMax {
+				t.Logf("The average is between the maximum and minimum values")
+			} else {
+				t.Fatalf("The average is not between the maximum and minimum values, Avg:%v Max:%v Min:%v", volAvg, volMax, volMin)
+			}
+
+			// Wait 120 sec cooling off period
```

Comment:
Maybe just `Wait for the cooling off period`, since the constants value could change.

Reply:
done